sherpa/ui/utils.py

```diff
@@ -5630,60 +5614,59 @@ def get_model(self, id=None):
-                                str_val, str_min, str_max = input.split(',')
-
-                                if str_val != "":
-                                    val = float(str_val)
-                                if str_min != "":
-                                    min = float(str_min)
-                                if str_max != "":
-                                    max = float(str_max)
-                            except Exception as e:
-                                info("Please provide a float value; " + str(e))
-                                continue
-                        else:
-                            info("Error: Please provide a comma separated list" +
-                                 " of floats; e.g. val,min,max")
-                            continue
-
-                        try:
-                            self.set_par(par, val, min, max)
-                            break
-                        except Exception as e:
-                            info(str(e))
-                            continue
-                    else:
-                        break
+        if not self._paramprompt:
+            return
+
+        try:
+            get_user_input = raw_input
+        except NameError:
+            get_user_input = input
+
+        for par in pars:
+            while True:
+                inval = get_user_input("%s parameter value [%g] " %
+                                       (par.fullname, par.val))
+                if inval == "":
+                    break
+
+                val = None
+                minval = None
+                maxval = None
+                tokens = [t.strip() for t in inval.split(',')]
+                ntokens = len(tokens)
+
+                if ntokens > 3:
+                    info("Error: Please provide a comma-separated " +
+                         "list of floats; e.g. val,min,max")
+                    continue
+
+                if tokens[0] != '':
+                    try:
+                        val = float(tokens[0])
+                    except Exception as e:
+                        info("Please provide a float value; " + str(e))
+                        continue
+
+                if ntokens > 1 and tokens[1] != '':
+                    try:
+                        minval = float(tokens[1])
+                    except Exception as e:
+                        info("Please provide a float value; " + str(e))
+                        continue
+
+                if ntokens > 2 and tokens[2] != '':
+                    try:
+                        maxval = float(tokens[2])
+                    except Exception as e:
+                        info("Please provide a float value; " + str(e))
+                        continue
+
+                try:
+                    self.set_par(par, val, minval, maxval)
+                    break
+                except Exception as e:
+                    info(str(e))
+                    continue
```

Comment:
I have a strong urge to refactor this code. It's probably not worth the effort, but that's what we say every time we see code like this... maybe we should start building some muscle memory for this kind of refactorings. @DougBurke what do you think?

Reply:
Well, as my code is a minor refactoring of the existing code, I can't really complain about it. I tried to make mine a minimal change to the old logic (to make it easier to follow), coupled with some test-coverage related changes (to make sure all the code was tested).

Given that before this test we don't have any tests of this logic, my suggestion would be to hold off refactoring this code in this commit. Then we have some test code that we have some confidence in to test out your refactoring. Maybe it would be worth removing my refactoring of this routine from this commit to make this a bit more obvious?

Reply:
Nah, I am OK with your changes, not with the underlying code. However, I am spending more time that I should on trying to lay out some work for important things like testing plots (well, not today) and testing log output. In any case I checked the test coverage and some of the error cases, as I think you mentioned in the comments, are not exercised, so at the very least I would try and add those tests in. Thing is, if we don't refactor it now we'll never refactor it as we'll be caught in other stuff.